xshop/pages/tests/test_views.py

```diff
@@ -13,3 +13,15 @@ def test_home_uses_desired_template(self):
 
         self.assertEqual(resp.status_code, 200)
         self.assertTemplateUsed(resp, "pages/home.html")
+
+
+class LoginTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+        self.url = reverse("login")
+
+    def test_login_uses_desired_template(self):
+        resp = self.client.get(self.url)
+
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, "registration/login.html")
```

Comment:
add a test case for invalid login,

help:
https://stackoverflow.com/questions/22457557/how-to-test-login-process